layouts/partials/navigation_level.html

```diff
@@ -42,22 +42,28 @@
             {{ end }}
 
             {{ if not .IsPage }}
-                <li role="treeitem" aria-label="{{ $linktitle}}">
+                <li role="treeitem" aria-label="{{ $linktitle }}">
                     {{ $collapse := not (.IsAncestor $current) }}
                     {{ if eq . $current }}
                         {{ $collapse = true }}
                     {{ end }}
 
-                    {{if lt (len $pageLocation) 6 -}}
-                        <a class='main' title="{{ $desc }}" href="{{ .Permalink }}">{{ $linktitle}}</a>
-                        {{ partial "navigation_level.html" (dict "pages" $pages "parent" . "current" $current "collapse" false "top" false "labelledby" "" ) .CurrentSection }}
-                    {{ else if (.CurrentSection.IsAncestor $current) }}
-                        <button {{ if not $collapse }} class="show" {{ end }} aria-hidden="true" tabindex="-1" ></button><a title="{{ $desc }}" href="{{ .Permalink }}">{{ $linktitle}}</a>
-                        {{ partial "navigation_level.html" (dict "pages" $pages "parent" . "current" $current "collapse" $collapse "top" false "labelledby" "" ) }}
-                    {{- else -}}
-                        <button {{ if not $collapse }} class="show" {{ end }} aria-hidden="true" tabindex="-1" ></button><a title="{{ $desc }}" href="{{ .Permalink }}">{{ $linktitle}}</a>
-                        {{ partialCached "navigation_level.html" (dict "pages" $pages "parent" . "current" $current "collapse" $collapse "top" false "labelledby" "" ) .CurrentSection }}
-                    {{- end -}}
+                    {{ if lt (len $pageLocation) 6 }}
+                        <a class='main' title="{{ $desc }}" href="{{ .Permalink }}">{{ $linktitle }}</a>
+                        {{ if and .CurrentSection (ne .CurrentSection .) (ne .CurrentSection $parent) (ne .CurrentSection $current) .CurrentSection.Pages }}
```

Comment:
Could you summarise the logic for me please?

basically, for each partial, I'm seeing "if all these things are true" - not root, not parent, not current etc.  Why both `.CurrentSection` and `.CurrentSection.Pages`?